components/Dialogs/src/MDCDialogTransition.m

```diff
@@ -45,11 +45,12 @@ - (instancetype)init {
   if (_presentationController) {
     return _presentationController;
   }
-  _presentationController =
+  MDCDialogPresentationController *presentationController =
```

Comment:
How is this block different?  Is the __weak annotation above sufficient for our needs?

Previously:
_presentationController = XYZ;

With this change:
UIPresentationController *localPresentationController = XYZ;
_presentationController = localPresentationController;

Reply:
I think I understand the need for a local variable.

Now that the ivar is weak, if it's the only one holding the reference to the newly-allocated `MDCDialogPresentationController`, it will immediately be deallocated. A strongly-retained local variable is required to get the instance back to the caller in the return statement.

Reply:
Ah, we are returning presentationController, not _presentationController.
Disregard.